### DISCUSSION TITLE
How can I save and get a saved object

### DISCUSSION THREAD

**Discussion by quick007** [Automattic/mongoose#10752] [Q&A] [answered]
Hi, I'm trying to be able to make a model then save that model. Heres my code: 

```js
let Homepage: mongoose.Model<mongoose.Document<any>>;

  if (!mongoose.models.Homepage) {
    const homepageSchema = new Schema({
      configured: Boolean,
        forums: {
          type: Map,
          of: Object
        },
        description: String,
        store: Boolean,
        storeLink: String,
        website: Boolean,
        websiteLink: String,
        custom: Boolean,
        customName: String,
        customLink: String, 
        orgName: String
    })
    Homepage = mongoose.model('Homepage', homepageSchema)
  }
  const orgName = new Homepage({ configured: false });
  await orgName.save()
  
  return res.status(200).json(await mongoose.get('Homepage'))
```

It's erroring all over the place (but no vscode errors strangely). How can I accomplish this? Also if you can show me a way where I can overwrite that, that would be great. (Like where I can plug in json values and it would edit the value in the schema, like `db.set( {description: "test"} )`)

**Comment by vkarpov15** [comment, marked as answer]
What are you trying to do with `await mongoose.get('Homepage')`? `mongoose.get()` is a function for getting the current value of a global object.

Maybe you're just trying to do `return res.status(200).json(orgName)`, or `return res.status(200).json(await Homepage.findById(orgName))`?

**Reply by quick007** [reply]
> What are you trying to do with `await mongoose.get('Homepage')`? `mongoose.get()` is a function for getting the current value of a global object.
> 
> Maybe you're just trying to do `return res.status(200).json(orgName)`, or `return res.status(200).json(await Homepage.findById(orgName))`?

This is what I want to do, how can I make a global object? I can't find any docs on it.

**Reply by quick007** [reply]
@vkarpov15 How to I make a global json object?

**Reply by quick007** [reply]
Nevermind, thanks all!

**Comment by quick007** [comment]
That works, but it keeps spamming my database (see image below). How can I save something then get back that same thing I saved in the database?

Heres my code:
```js
  Homepage = mongoose.model('Homepage', homepageSchema)
  const orgName = new Homepage({ configured: true });
  await orgName.save()
  
  return res.status(200).json(await Homepage.findById(orgName))
```

![image](https://user-images.githubusercontent.com/69743171/135568204-6228a7f5-6e56-4070-9373-72b283060980.png)

**Reply by vkarpov15** [reply]
`findOneAndUpdate()`:

https://mongoosejs.com/docs/tutorials/findoneandupdate.html

**Comment by imduchy** [comment]
You should be able to do

```typescript
const orgName = await Homepage.create({ configured: true })
return res.status(200).json(orgName)
```

Reference: https://mongoosejs.com/docs/api.html#model_Model.create
